docs/dev/generator.js

```diff
@@ -53,21 +53,51 @@ const fail = (...args) => {
   if ( forReal ) { hasError = true; }
 };
 
+
+const languages = [
+  ...shiki.BUNDLED_LANGUAGES.map((l) => {
+    if ( l.id === 'javascript' ) {
+      l.aliases.push('mjs');
+    }
+    return l;
+  }),
+  {
+    id: 'reach',
+    scopeName: 'source.js',
+    // XXX customize this
+    path: 'languages/javascript.tmLanguage.json',
+    aliases: ['rsh'],
+  },
+];
+const languageNormalizeDict = {};
+for (const l of languages) {
+  languageNormalizeDict[l.id] = l.id;
+  for (const alias of (l.aliases || [])) {
+    languageNormalizeDict[alias] = l.id;
+  }
+}
+const normalizeScope = (s) => languageNormalizeDict[s] || s;
```

Comment:
I think you should error if there's no normalization and we'd fix it

Reply:
So one thing that that allows through is that a bunch of things are added with a literal 'h' scope.  Also with literal 'term' scope.  These are not languages that Shiki knows about.  So this cross reference stuff is used for both code cross references and document cross references.

Reply:
Shall we add a list of literals like 'h' and 'term' to the normalizer, and error if it's not a language name in the shiki list or a known document scope literal?

Reply:
I see. Yea, I think you should add them.